tsl/org.testeditor.tsl.dsl.tests/src/org/testeditor/tsl/dsl/parser/MarkupParserTest.xtend

```diff
@@ -0,0 +1,146 @@
+package org.testeditor.tsl.dsl.parser
+
+import org.eclipse.xtext.junit4.XtextRunner
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.testeditor.tsl.util.TslModelUtil
+import com.google.inject.Inject
+import org.testeditor.tsl.TextLine
+import org.testeditor.tsl.BoldText
+import org.testeditor.tsl.BoldCursiveText
+import org.testeditor.tsl.CursiveText
+
+@RunWith(XtextRunner)
+class MarkupParserTest extends AbstractParserTest {
+
+	@Inject extension TslModelUtil
+
+	@Test
+	def void missingPackage(){
+		// given
+		val tsl = '''
+			// package packageA
+		'''
+		
+		// expect nothing parsed
+		tsl.parse[
+			assertNull	
+		]
+	}
+	
+	@Test
+	def void packageAlone(){
+		// given
+		val tsl= '''
+			package packageA
+		'''
+
+		// expect		
+		tsl.parse[
+			package.assertEquals( "packageA" )
+		]
+	}
+
+	@Test
+	def void packageWithComments(){
+		// given
+		val tsl='''
+			// comment
+			/* comment
+			     */package/* ohoh */packageA// tata
+			     /*
+			*/
+		'''
+		
+		// expect
+		tsl.parse[
+			package.assertEquals( "packageA" )
+		]
+	}
+	
+	@Test
+	def void specWithStepsAndHeadlineMarkups(){
+		// given
+		val tsl='''
+			package packageB
+			# TestSpecName
+			
+			some markup
+			===========
+			
+			single underlined
+			-----------------
+			
+			other not underlined
+			
+			* step1_1 "value"     /*test */ step1_2 step1_3
+				// inter line 
+			  * step2_1.
+			 *    step3_1  step3_2 
+		'''
+		
+		// expect 
+		tsl.parse[
+			specification.steps.assertSize(3) => [
+					get(0).contents.restoreString.assertEquals('step1_1 "value" step1_2 step1_3')
+					get(1).contents.restoreString.assertEquals('step2_1')
+					get(2).contents.restoreString.assertEquals('step3_1 step3_2')
+				]			
+			specification.descriptions.assertSize(3) => [
+				get(0) => [
+					textline.elements.map[ text ].join(" ").assertEquals("some markup")
+					doubleUnderlined.assertTrue				
+					underlined.assertFalse
+				]
+				get(1) => [
+					textline.elements.map[ text ].join(" ").assertEquals("single underlined")
+					doubleUnderlined.assertFalse
+					underlined.assertTrue
+				]
+				get(2) => [
+					textline.elements.map[ text ].join(" ").assertEquals("other not underlined")
+					doubleUnderlined.assertFalse
+					underlined.assertFalse
+				]
+			]
+		]
+	}
+
+	@Test 
+	def void boldAndCursiveMarkup() {
+		// given
+		val tsl = '''
+			package test
+			
+			# Test
+			 
+			 
+			  1 + 2 = 33, +bold+ t // test				  
+			  some issue				  
+			    1. +important stuff+ ++cursive++ +++matter+++ that + 4 does //				    
+			    2. compute
+			    /*
+			  ok
+			  */
+			* ok.  
+		'''
+		
+		// expect
+		tsl.parse[
+			specification.steps.assertSize(1)
+			specification.descriptions.assertSize(4) => [ // four markup lines
+				get(0).textline.elements.assertSize(8) => [ // eight elements in first markup line
+					(get(6).me as BoldText).text.head.assertEquals('bold')					

```

Comment:
You could use `get(6).assertInstanceOf(BoldTest)` instead here. If `get(6)` is not of type `BoldText` the assertion message will be easier to understand than the otherwise raised `ClassCastException`.